### ISSUE TITLE
Toolbar Items with text using different font colors on Windows and Mac

### ISSUE THREAD

**Issue by tobiasmelcher** [eclipse-platform/eclipse.platform.swt#1696]
Following SWT code snippet is creating a label and a toolbar with one item.

```
import org.eclipse.swt.SWT;
import org.eclipse.swt.layout.GridData;
import org.eclipse.swt.layout.GridLayout;
import org.eclipse.swt.widgets.Display;
import org.eclipse.swt.widgets.Label;
import org.eclipse.swt.widgets.Shell;
import org.eclipse.swt.widgets.ToolBar;

import org.eclipse.jface.action.Action;
import org.eclipse.jface.action.ActionContributionItem;
import org.eclipse.jface.action.ToolBarManager;
import org.eclipse.jface.resource.ImageDescriptor;

public class ToolbarExample {
	public static void main(String[] args) {
		Display display= new Display();

		Shell shell= new Shell(display);
		shell.setText("Toolbar Example");
		shell.setLayout(new GridLayout(1, false));

		Label label= new Label(shell, SWT.NONE);
		label.setText("Label");

		var tm= new ToolBarManager(SWT.FLAT | SWT.HORIZONTAL | SWT.RIGHT);
		var action= new Action("Toolbar Action") {
			@Override
			public void run() {
			}
		};
		action.setImageDescriptor(ImageDescriptor.createFromImage(display.getSystemImage(SWT.ICON_INFORMATION)));
		var aci= new ActionContributionItem(action);
		aci.setMode(ActionContributionItem.MODE_FORCE_TEXT);
		tm.add(aci);

		ToolBar fToolBar= tm.createControl(shell);
		GridData gd= new GridData(SWT.BEGINNING, SWT.BEGINNING, false, false);
		fToolBar.setLayoutData(gd);

		shell.pack();
		shell.open();

		while (!shell.isDisposed()) {
			if (!display.readAndDispatch()) {
				display.sleep();
			}
		}
		display.dispose();
	}
}
```

The result looks different on Windows on Mac:

Windows:
![toolbar_windows](https://github.com/user-attachments/assets/8f9d4eb7-f84d-454f-90e6-f04c64639240)

Mac:
<img width="125" alt="toolbar_mac" src="https://github.com/user-attachments/assets/ade155cb-b2cd-496d-b1c1-9978f4be5611" />

Why is using the Mac SWT implementation a different font color for the toolbar item compared to Windows? For the label, both implementations are using a black font. 
Are all toolbar item texts on Mac rendered with a gray-isch font? So, is it intended that toolbars are rendered differently with a native look and feel on each platform?
I personally think that the toolbar item texts should be rendered with the same color on each platform. What do you think?

**Comment by BeckerWdf** [comment]
So in the IDE we can see this e.g. in the "perspective switcher" (when the "show text" option is turned on), correct?

**Comment by tobiasmelcher** [comment]
yes, exactly, here some more screenshots from the perspective switcher.

Windows Light Mode:
![windows_light](https://github.com/user-attachments/assets/c1cc4b70-bec4-448d-a793-02533f9c148d)

MacOS Light Mode:
<img width="333" alt="mac_light" src="https://github.com/user-attachments/assets/84e1b90e-6e49-44d1-8773-36c0330b430d" />

Windows Dark Mode:
![windows_dark](https://github.com/user-attachments/assets/f2d19244-1723-41d0-aece-6b2df34466a2)

Black color is used as foreground for the View title text (Outline, Variabl...) on both platforms consistently; but not for the texts in the perspective switcher (Debug, Java). Why?


MacOS Dark Mode:
<img width="335" alt="mac_dark" src="https://github.com/user-attachments/assets/16998d58-f7f2-4770-af0d-62b8bb7e522e" />

In dark mode both platforms use a white color as foreground in the perspective switcher texts. (On windows, the text cannot be read in dark mode, but this is worth for another issue) 
So, why using different foreground colors in light mode on Windows and MacOS but not in dark mode?

**Comment by BeckerWdf** [comment]
> In dark mode both platforms use a white color as foreground in the perspective switcher texts. (On windows, the text cannot be read in dark mode, but this is worth for another issue)

I think it's this: https://github.com/eclipse-platform/eclipse.platform.ui/issues/2570

**Comment by MikeMCrank** [comment]
I've been exploring possible solutions for #2570, looks like I have a bit more work to do in order to handle text in toolbar buttons.
